### PR TITLE
form: add the possibility to order a select menu

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Adds a new SelectWithOrder type for ngx-formly with order parameter.
* Adds a new pipe to user sort options into the template.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Why are you opening this PR?

- https://tree.taiga.io/project/rero21-reroils/task/1528

## How to test?

- Create a new document and check the order on Select Menu

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/rero/ng-core/src/lib/record/editor/type/select-with-sort-type.component.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  /**
+   * Menu order (Default order: label, desc ex: '-label')
+   */
+  public order = 'label';
+
+  /** Constructor */
+  constructor() {
+    super();
+  }
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/editor/type/select-with-sort-type.component.ts:44]: I think you can omit the constructor and the super will be called implicitly.

Comment (Garfield-fr) [reply projects/rero/ng-core/src/lib/record/editor/type/select-with-sort-type.component.ts]: done

File: projects/rero/ng-core/src/lib/record/editor/type/select-with-sort-type.component.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+export class SelectWithSortTypeComponent extends FieldType implements OnInit {
+  /**
+   * Menu order (Default order: label, desc ex: '-label')
+   */
+  public order = 'label';
+
+  /** Init */
+  ngOnInit() {
+    const templateOptions = this.field.templateOptions;
+    if ('selectWithSortOptions' in templateOptions) {
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/editor/type/select-with-sort-type.component.ts:43]: you can replace the two `if` by a `&&`.

File: projects/rero/ng-core/src/lib/pipe/sort-by-keys.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import { sortByKeys } from '../utils/sort-by-keys';
+
+/**
+ * Pipe for the sorting function.
+ */
+@Pipe({
+    name: 'sortByKeys'
+})
+export class SortByKeysPipe implements PipeTransform {
+  public transform(value: any, ...keys: string[]): any[] {
```
Comment (jma) [inline projects/rero/ng-core/src/lib/pipe/sort-by-keys.pipe.ts:29]: documentation is missing.

File: projects/rero/ng-core/src/lib/record/editor/type/select-with-sort-type.component.ts (hunk 2 above)
Comment (Garfield-fr) [reply projects/rero/ng-core/src/lib/record/editor/type/select-with-sort-type.component.ts:43]: In the near future, we will have another set of possible values

File: projects/rero/ng-core/src/lib/pipe/sort-by-keys.pipe.ts (hunk 3 above)
Comment (Garfield-fr) [reply projects/rero/ng-core/src/lib/pipe/sort-by-keys.pipe.ts:29]: done

**Pull request merged**